### ISSUE TITLE
XOT Gap Fill issues

### ISSUE THREAD

**Issue by JohnSmith-LT** [thexerteproject/xerteonlinetoolkits#916]
Alongside the Drag Drop page there are at least 4x open issues with the Gap Fill page:

- issue with audio on gap fill page if both narration and audio feedback options are added Page types XOT template bug futureproofing - https://github.com/thexerteproject/xerteonlinetoolkits/issues/813

- Gap Fill answer incorrectly treated as wrong after going over pages using next/back buttons - https://github.com/thexerteproject/xerteonlinetoolkits/issues/811

- Gap Fill answer incorrectly becomes red when going over pages using next/back buttons - https://github.com/thexerteproject/xerteonlinetoolkits/issues/810

- Gap Fill with 2 Fill in Blanks, hangs when going over pages using next/back buttons - https://github.com/thexerteproject/xerteonlinetoolkits/issues/807

Not sure at what stage any of them are but it seems like this and the DragDrop are being used significantly enough that we should get them back to working reliably.

Like the DragDrop one most of the problems seem to be in the interoperability between two pages of the same type since revisiting a page isn't mainitaining the previous state as variables have been overwritten by similar pages.

**Comment by ronm123** [comment]
@JohnSmith-LT just catching up on some of this and for now just a simple comment that I think a lot of testing that I/we do is using a page types demonstrator with one of each of the page types after the other, or a new project with a particular page type added amongst other page types, but rarely a sequence of the same page type. I think in normal development there is also a tendency to vary the page types used rather than one after the other of the same page type so perhaps this all explains why we don't get more regular reports of these issues. Look forward to discussing further tomorrow.

**Comment by ronm123** [comment]
@JohnSmith-LT @FayCross @torinfo re gap fill and correct/incorrect feedback. I'm adding this here because I can't find where we've discussed this before even though I know we definitely have discussed it before!

During the accessibility testing Alistair reported that the gap fill in drop down mode only uses colour to indicate which answers are correct e.g. green which doesn't work for colour blind or partially sighted users. I know we've looked at this before and in our page types demonstrator I added some css as a possible solution e.g. https://xot.xerte.org.uk/play.php?template_id=116#gapfilldropdown which keeps the green but also includes a dashed border too.
I added similar css to the drag and drop version https://xot.xerte.org.uk/play.php?template_id=116#gapfilldraganddrop but noticed that doesn't work correctly when forced tracking mode is turned on.
However it works fine in text entry mode regardless of force tracking mode https://xot.xerte.org.uk/play.php?template_id=116#gapfillfillintheblank

The questions is do we go with this as an option and fix whatever is wrong with the drag and drop mode when force tracking is on or do we aim for better consistency with other page types where ticks and crosses are used to indicate correct/incorrect? I'm in two minds because the ticks might make the display in one or more of these drop down modes a bit busy and confusing. On the other hand I think more consistency would be good. Looking at the pages where we currently include ticks and/or crosses I made the following notes. These also include notes about the difference between with and without force tracking and whether we have a restart and/or continuous interaction etc in either mode but I guess that should be a separate discussion. The main point relevant here is use of ticks and crosses...

answer x of y = on click of check button tick or cross displayed in feedback if all correct or not but not agaisnt individual responses + no restart button
categories = ticks and crossess shown against individual responses + no need for restart button but should there be?
dialogue = ticks shown when answer correct + no restart button because hints mean always correct but should there be an option to not show hints?
gap fill drag and drop = no ticks and green/dotted feedback intermittant when force tracking is on - should be ticks anyway?
gap fill drop down = on click of check button drop down green dotted via custom css but should be ticks?
gap fill fill in the blank - green and borders works even in force tracking mode but should it be ticks?
matching texts = on click of submit button ticks and crosses show against each item in force tracking mode. Without force tracking no crosses but incorrect are put back and all items can be moved continuously.
sortable grid = on click of check answers ticks and crosses shown without force tracking continuous interaction possible. With force tracking check button removed.
text correction tick or cross displayed in feedback
timeline/matching pairs = on click of check answers ticks and crossess shown. without force tracking continuous interaction. With force tracking check button disappears.
